src/components/shared/formItems/__tests__/EthAddress.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react'
+
+import userEvent from '@testing-library/user-event'
+
+import { useForm } from 'antd/lib/form/Form'
+
+import { Form } from 'antd'
+
+import { render } from '../../../../../test/test-utils'
+import EthAddress from '../EthAddress'
+import { readProvider } from '../../../../constants/readProvider'
+
+jest.mock('../../../../constants/readProvider')
```

Comment:
I don't love this, but it can work for us in a limited capacity until there's a sort-of mocked chain we can lean on. It can be spun up in `src/setupTests.ts` in `beforeEach` (or `beforeAll` if it can be completely reset within `beforeEach`)

Reply:
I think this is totally valid; indeed, ideal. From the perspective of this component, `readProvider`'s behavior is irrelevant. As long as `EthAddress` is calling the correct things, `readProvider`'s behavior can be independently tested.